rust/src/database/verkle/variants/managed/nodes/mod.rs

```diff
@@ -43,12 +46,22 @@ pub enum Node {
     Inner(Box<InnerNode>),
     Leaf2(Box<Leaf2Node>),
     Leaf256(Box<Leaf256Node>),
+    // Make sure to adjust smallest_leaf_type_for when adding new leaf types.
 }
 
 type Leaf2Node = SparseLeafNode<2>;
 type Leaf256Node = FullLeafNode;
 
 impl Node {
+    /// Returns the smallest leaf node type capable of storing `n` values.
+    pub fn smallest_leaf_type_for(n: usize) -> NodeType {
+        match n {
+            0..=2 => NodeType::Leaf2,
```

Comment:
Although never called, maybe for consistency
```suggestion
            0 => NodeType::Empty,
            1..=2 => NodeType::Leaf2,
```